torch/csrc/jit/tensorexpr/loopnest.cpp

```diff
@@ -1911,6 +1911,58 @@ std::vector<For*> LoopNest::reorder(
   return result;
 }
 
+std::vector<For*> LoopNest::tile(For* x, For* y, int x_factor, int y_factor) {
+  auto parent = dynamic_cast<Block*>(x->get_parent());
+  if (parent == nullptr) {
+    throw malformed_input("parent of the loops must be a Block");
+  }
+  if (!areLoopsPerfectlyNested({x, y})) {
+    throw malformed_input("two loops must be perfectly nested");
+  }
+
+  // Split x, y axes by x_factor and y_factor
+  For *yi, *ytail;
+  splitWithTail(y, y_factor, &yi, &ytail);
+  For *xi, *xtail;
+  splitWithTail(x, x_factor, &xi, &xtail);
+
+  // Distribute xi over yo and ytail so we can manipulate the loop order of {xo,
+  // xi, yo, yi}
+  auto loops = distributeLoop(xi);
+
+  // For {xo, xi, yo, yi}, reorder the axes to be xo, yo, xi, yi
+  xi = loops.front();
+  For* yo = dynamic_cast<For*>(xi->body()->stmts().front());
+  CHECK(yo);
+  reorderAxis(xi, yo);
+
+  // Re-Identify loop handles of yo, xi, yi, ytail
+  For* xo = x;
+  yo = dynamic_cast<For*>(xo->body()->stmts().front());
+  CHECK(yo);
+  xi = dynamic_cast<For*>(yo->body()->stmts().front());
+  CHECK(xi);
+  yi = dynamic_cast<For*>(xi->body()->stmts().front());
+  CHECK(yi);
+
+  if (ytail) {
+    ytail = dynamic_cast<For*>(xo->body()->stmts().back());
+    CHECK(ytail);
+    ytail = dynamic_cast<For*>(ytail->body()->stmts().front());
+    CHECK(ytail);
+  }
+
```

Comment:
Don't you have to fuse the loops, since you distribute above? The documentation in the `.h` file says that the output will be fused.

Reply:
It's already fused and the structure is like the following,
```
xo
  yo
    xi
      yi
  xi
    ytail
xtail
    
```

Reply:
I was referring to the distribute above. But I get your point now. Since you are only distributing `xi` and that is reordered with `yo`, there is nothing to fuse here.